src/mlpack/tests/knn_test.cpp

```diff
@@ -245,6 +245,7 @@ BOOST_AUTO_TEST_CASE(TrainTreeTest)
   BOOST_REQUIRE_EQUAL(neighbors.n_cols, baselineNeighbors.n_cols);
   BOOST_REQUIRE_EQUAL(distances.n_rows, baselineDistances.n_rows);
   BOOST_REQUIRE_EQUAL(distances.n_cols, baselineDistances.n_cols);
+  BOOST_REQUIRE_EQUAL(oldFromNewReferences.size(), distances.n_cols);
```

Comment:
It's a little irritating but I guess to fix the static code analysis issue we can replace `i < distances.n_cols` with `i < oldFromNewReferences.size()`, since those should be the same anyway.

Reply:
It is :/ It got fixed with f35b22d38f3e94a16011e9caa1c5a6aaa849db39.